weave

```diff
@@ -532,21 +533,28 @@ dns_args() {
     DNS_ARGS="$DNS_ARG $DNS_SEARCH_ARG"
 }
 
-# Perform operation $1 on container ID $2 to local DNS database at address $3
+# Perform operation $1 on container ID $2 to local DNS database with addresses at $3
+# followed by an optional fully-qualified domain name
 # This function is only called where we know $2 is a valid container name
 tell_dns() {
     METHOD="$1"
     CONTAINER_ID="$2"
     shift 2
-
+    collect_cidr_args "$@"
+    shift $CIDR_COUNT
```

Comment:
@rade heh, I actually _had_ done it that way, then I refactored to what I submitted. I think you'er right, though, it would be better as two separate functions.